### PR TITLE
CircularProgress: typing updates

### PR DESCRIPTION
**Pull request by danethurber**
@jhewlett let me know what you think. i was attempting to make the component typings simpler to read and replicate. i had some patterns that i've been using on another project that i wanted to run by you

### CODE REVIEW

File: packages/circularprogress/src/index.d.ts [hunk 1]
```diff
@@ -1,24 +1,39 @@
-import { ForwardRefExoticComponent, RefAttributes } from 'react'
+import { ForwardRefExoticComponent, HTMLAttributes, RefAttributes } from 'react'
 
-interface Styling {
-  css?: Record<string, string | number>
```
Comment (danethurber) [inline packages/circularprogress/src/index.d.ts:4]: `css` isn't needed anymore. this is something we should have removed when we dropped glamourous

File: packages/circularprogress/src/index.d.ts [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
-  style?: Record<string, string | number>
-  className?: string
```
Comment (danethurber) [inline packages/circularprogress/src/index.d.ts:6]: these are react/dom native attrs and we shouldn't have to define them for all components. we can leverage the `HTMLAttributes` typing to get the list of this props. if we were to do this is our react source we wouldn't need the filterReactProps util anymore

File: packages/circularprogress/src/index.d.ts [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
+  HTMLDivElement,
+  Statics
+>
+
+export default CircularProgress
 
-export interface CircularProgressProps {
-  size?: 'small' | 'medium'
+export interface Props {
+  size?: keyof typeof sizes
```
Comment (danethurber) [inline packages/circularprogress/src/index.d.ts:12]: generate the types from the sizes src/const

File: packages/circularprogress/src/index.d.ts [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
+  Props,
+  HTMLDivElement,
+  Statics
+>
+
+export default CircularProgress
 
-export interface CircularProgressProps {
-  size?: 'small' | 'medium'
+export interface Props {
```
Comment (jhewlett) [inline packages/circularprogress/src/index.d.ts:11]: The reason I named this `CircularProgressProps` is to follow a convention of `{ComponentName}Props`, which would provide users with a consistent way to augment the props interface if they need to, like so:

```typescript
declare module '@pluralsight/ps-design-system-tab' {
    interface ListItemProps {
        name: string                          //add "name" prop to ListItem
    }
}
```

It feels redundant with single-export components, but makes more sense when you consider things like `Tab` that have multiples, and this way there would only be one convention to remember.

That said, now that you've added the `HTMLAttributes` in to the props, does that cover all the props that a user might provide (and would actually get forwarded on?) If so, then there's probably less need to augment, and I don't really care either way.

File: packages/circularprogress/src/index.d.ts [hunk 5]
```diff
@@ -1,24 +1,39 @@
...
-  small: 'small',
-  medium: 'medium'
+export interface Statics {
+  sizes: typeof sizes
 }
+export const sizes = keyMirror('small', 'medium')
 
-export const sizes: Sizes
+// TODO: relocate to a global typings file
+export type RefForwardingComponent<
```
Comment (jhewlett) [inline packages/circularprogress/src/index.d.ts:22]: I really like this helper!

File: packages/circularprogress/src/index.d.ts (hunk 4 above)
Comment (jaketrent) [reply packages/circularprogress/src/index.d.ts:11]: I like the `{ComponentName}Props` convention.

**Pull request merged**